presto-spi/src/main/java/com/facebook/presto/spi/type/NamedTypeSignature.java

```diff
@@ -25,20 +25,32 @@
 public class NamedTypeSignature
 {
     private final Optional<String> name;
+    private final boolean delimited;
     private final TypeSignature typeSignature;
 
     @JsonCreator
     public NamedTypeSignature(
             @JsonProperty("name") Optional<String> name,
+            @JsonProperty("delimited") boolean delimited,
             @JsonProperty("typeSignature") TypeSignature typeSignature)
     {
         this.name = requireNonNull(name, "name is null");
+        if (!name.isPresent() && delimited) {
+            throw new IllegalArgumentException("delimited but name is absent");
+        }
+
+        this.delimited = delimited;
         this.typeSignature = requireNonNull(typeSignature, "typeSignature is null");
     }
 
     public NamedTypeSignature(String name, TypeSignature typeSignature)
```

Comment:
Remote this overload.

In general, we don't have overloads like this. Doing it here is particularly bad because the omittable argument is in the middle.